content/en/docs/features/monitoring-and-diagnostics/monitoring-device-connectivity.md

```diff
@@ -0,0 +1,192 @@
+   deactivate CoAPGateway
+   loop for each offline service
+      loop for each device associated with the offline service
+         ResourceAggregate -> ResourceAggregate: Send UpdateDeviceMetadataRequest with\nDevice ID, Connection.Status = OFFLINE
+         ResourceAggregate -> EventBus: Publish DeviceMetadataUpdated with\nConnection.Status = OFFLINE, Service.Id = ""
+      end
+      ResourceAggregate -> ResourceAggregate: Confirm the offline service
+   end
+   deactivate ResourceAggregate
+end
+
+== Device Online ==
+
+CoAPGateway -> ResourceAggregate++ : Send UpdateDeviceMetadataRequest with\nDevice ID, Connection.Status = ONLINE, Service.Id = "ID"
+ResourceAggregate -> EventBus: Publish DeviceMetadataUpdated with\nConnection.Status = ONLINE, Service.Id = "ID"
+return UpdateDeviceMetadataResponse
+
+@enduml
+{{< /plantuml >}}
+
+### Example
+
+This series of steps describes the process of expanding the setup, encountering an Out of Memory event, recording and analyzing gateway states, and identifying gateway termination within a Kubernetes environment using CoAP gateways and the Resource Aggregate.
+
+1. **Initial Setup Expansion:**
+   - Added two CoAP gateways to the Kubernetes (k8s) environment.
+   - Configured a 1-minute time to live, so each CoAP gateway will update the database every minute.
+   - Current time is `Wed Sep 27 2023 13:47:20 GMT+0000`.
+   - In the database, there are two CoAP gateways in the online state:
+
+     ```jsonc
+     {
+        "online": [
+           {
+              "id": "ID-0",
+              "onlineValidUntil": "1695822490000000000" // Wed Sep 27 2023 13:48:10 GMT+0000
+           },
+           {
+              "id": "ID-1",
+              "onlineValidUntil": "1695822500000000000" // Wed Sep 27 2023 13:48:20 GMT+0000
+           }
+        ],
+        "offline": []
+     }
+     ```
+
+2. **OOM Event and Restart:**
+   - One gateway instance (ID-1) experiences an Out of Memory (OOM) event and restarts.
+   - Two CoAP gateway instances exist concurrently (ID-0 and ID-2), while ID-1 is dead but still in the online state.
+   - Current time is `Wed Sep 27 2023 13:47:30 GMT+0000`.
+
+     ```jsonc
+     {
+        "online": [
+           {
+              "id": "ID-0",
+              "onlineValidUntil": "1695822490000000000" // Wed Sep 27 2023 13:48:10 GMT+0000
+           },
+           {
+              "id": "ID-1", // Dead instance
+              "onlineValidUntil": "1695822500000000000" // Wed Sep 27 2023 13:48:20 GMT+0000
+           },
+           {
+              "id": "ID-2", // New instance
+              "onlineValidUntil": "1695822510000000000" // Wed Sep 27 2023 13:48:30 GMT+0000
+           }
+        ],
+        "offline": []
+     }
+     ```
+
+3. **Update record by CoAP gateway (ID-0):**
+   - After 20 seconds, the CoAP gateway (ID-0) updates its record.
+   - Current time is `Wed Sep 27 2023 13:47:50 GMT+0000`.
```

Comment:
The previous `onlineValidUntil` value for `ID-0` is `13:48:10`, so it should wake up after 20secs at `13:48:30` and not at `13:48:50`, no?

Reply:
No it is wake: `13:47:30(now)` + (`13:48:10(heartbeatValidUntil)`-`13:47:30(now)`)/3 ~= `13:47:50`